### DISCUSSION TITLE
Uno R4 boards

### DISCUSSION THREAD

**Discussion by Grumpy-Mike** [FortySevenEffects/arduino_midi_library#334] [General]
There are two new boards going under the name Uno R4. One is the Minima and the other is the WIFi.
While the Basic IO example will work on the Minima, when trying it on the WIFI it causes the system to crash and the board will no longer be able to upload anything, even the blink example, citing that:-
No device found on cu.usbmodemDC5475C45C282
Getting it back to work involves a lot of messing about that I have still not nailed down to a specific procedure.

Both Uno R4 boards work when you send MIDI "by hand", that is writing at MIDI baud rates to the appropriate port using Serial.write. 

I am assuming that there is something about the WIFI that the serial.begin doesn't like. I have studied the source code of the library and the documentation, but I can't seem to find anything.

I do appreciate that this might be a problem with the R4 board libraries themselves, but as it shows up on using the MIDI library I thought I would ask here as well. 

Thank you for reading.

**Comment by Grumpy-Mike** [comment]
I have managed to solve the problem. It was to do with the way the WIFI board switched the access to the USB as opposed to the Minima board.
It just involved adding two commands in the setup function. You can find the thread here
[https://forum.arduino.cc/t/midi-library-works-on-minima-but-not-on-wifi/1165783](url)

I will now continue looking at what is supported on these new boards.

**Reply by yashiquechalil** [reply]
This link seems to be broken. Would love to get MIDI working with my r4 wifi

**Reply by per1234** [reply]
Hi @yashiquechalil. The link above is broken due to the use of incorrect markup in the post source. This link will work for you:

https://forum.arduino.cc/t/midi-library-works-on-minima-but-not-on-wifi/1165783

**Reply by yashiquechalil** [reply]
I see, I did follow this thread to get here in the first place. 
Anyways, adding   pinMode(21, OUTPUT);
    digitalWrite(21, LOW);
to my setup() didn't really solve the problem. Hairless still does not read any messages from the board

**Reply by adictoetico** [reply]
I was having trouble using MIDI Input on a UNO R4 WIFI and ran into your posts about it, I tried using pin 21, but I didnt get  the results I needed, my solution was to edit the SerialMIDI.h file in the library to include (ARDUINO_UNOR4_WIFI) in the boards it checks to use Serial1 instead of Serial, that way i am communicating MIDI throught the RX and TX pins on the board. Hope this helps, as this posts also drove me in the right direction....
